README.md

```diff
@@ -1,340 +1,535 @@
+
+### DKG
+
+Distributed Key Generation is a cryptographic process that aims to solve the problem of coordinating N parties to cryptographically sign and verify signatures without relying on Trusted Third Parties. The process is demonstrated to be successful in computing a key pair in the presence of a number T attackers in a decentralized network. To do so, this algorithm generates a public key, and a secret key of which no single party knows, but has some share of. The involvement of many parties requires Distributed key generation to ensure secrecy in the presence of malicious contributions to the key calculation.
+For more information about DKG in general, [please visit this page](https://en.wikipedia.org/wiki/Distributed_key_generation).
+
+### DKG tool by SSV
+
+The SSV team built this tool leveraging [drand](https://drand.love/)'s DKG protocol implementation ([please visit their documentation](https://drand.love/docs/cryptography/#setup-phase) for more details on it). This implementation operates under the assumption of a p2p network, allowing operators to communicate.
+The `ssv-dkg` was built to lift this assumption and provide a communication layer that centered on an Initiator figure, to facilitate communication between operators. The introduced potential risk for centralization and bad actors is handled with signatures and signature verifications, as explained in the Security notes section.
+Finally, the outcome of the DKG ceremony is a BLS key pair to be used for validator duties by operators on the ssv.network. As such, the tool ends the process by creating a deposit file to activate the newly created validator key pair, and proceeds to generating the payload for the transaction, and triggering the transaction itself.
+
+## Overview
+
+In order for the DKG protocol to execute successfully:
+
+* all the chosen Operators must be running the `ssv-dkg` tool as Operators
+* separately, an Initiator (one of the Operators, or a separate entity), starts the DKG ceremony by running the ssv-dkg tool with the init parameter
+* the tool automatically exchange data between the interested parties, as outlined in the [Flow Description](#flow-description) section, until the key shares are created
+
+For details on how to run the tool as an Operator, please head over [to this section containing the related instructions](#operator-quick-start).
+Similarly, head over to [this other section](#initiator-quick-start) for instructions on how to launch the tool as the Initiator of the DKG ceremony.
+
+> ℹ️ NOTE: Threshold is computed automatically using 3f+1 tolerance.
+> 
+## Initiator Quick start
+
+### Generate Initiator identity RSA key pair
+
+The Initiator needs to sign all messages exchanged with DKG participants with an RSA key.
+First of all, write down your chosen password in a text file, for example `password`, replacing `<PASSWORD>` with a password of your choosing:
 
-### Build
+```sh
+echo "<PASSWORD>" >> password
```

Comment:
since we added the option to use an automatic generated key, I assume this will be the easiest and default option for most. I would structure this part like this - 

including the `--generateInitiatorKey` flag generates the key for you and saves it in the output folder.
IF you want to use your own key or generate please follow these instructions -
proceed to give instructions.

starting with how to create the key might confuse users that anyway want to use the easier faster option.

Reply:
I guess its missing the part about this option anyway.

Reply:
I agree, but I thought I'd add this at a later stage, not in this merge, since I didn't have time for testing this part  yesterday.